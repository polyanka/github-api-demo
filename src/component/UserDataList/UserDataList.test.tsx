import { shallow } from 'enzyme';
import * as React from 'react';

import { UserDataList } from './index';

describe('<UserDataList />', () => {
  const mockUser = {
    avatar_url: 'https://www.example.com/avatar_url.png',
    id: 1,
    login: 'Login',
    name: 'User',
    blog: 'https://www.example-blog.com',
    location: null,
    public_repos: 0,
    public_gists: 2,
    followers: 2,
    following: 0,
    company: null,
  };

  it('renders correctly UserDataList component', () => {
    const wrapper = shallow(<UserDataList user={mockUser} />);
    expect(wrapper).toMatchSnapshot();
  });
});
