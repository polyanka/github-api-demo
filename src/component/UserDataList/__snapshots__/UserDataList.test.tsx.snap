// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserDataList /> renders correctly UserDataList component 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <WithStyles(ForwardRef(CardMedia))
      image="https://www.example.com/avatar_url.png"
      style={
        Object {
          "height": 240,
          "width": "100%",
        }
      }
      title="Login"
    />
    <WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(Typography))
        color="primary"
        gutterBottom={true}
        variant="h5"
      >
        User
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        variant="h5"
      >
        Login
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(List))>
      <WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItemIcon))>
          <PeopleAltOutlinedIcon
            color="primary"
          />
        </WithStyles(ForwardRef(ListItemIcon))>
        <WithStyles(ForwardRef(ListItemText))
          primary="Followers"
          secondary={2}
        />
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItemIcon))>
          <EmojiPeopleOutlinedIcon
            color="primary"
          />
        </WithStyles(ForwardRef(ListItemIcon))>
        <WithStyles(ForwardRef(ListItemText))
          primary="Following"
          secondary={0}
        />
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItemIcon))>
          <StarBorderOutlinedIcon
            color="primary"
          />
        </WithStyles(ForwardRef(ListItemIcon))>
        <WithStyles(ForwardRef(ListItemText))
          primary="Start Repos"
          secondary={0}
        />
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListItem))
        button={true}
        component="a"
        href="https://www.example-blog.com"
        target="_blank"
      >
        <WithStyles(ForwardRef(ListItemIcon))>
          <LinkOutlinedIcon
            color="primary"
          />
        </WithStyles(ForwardRef(ListItemIcon))>
        <WithStyles(ForwardRef(ListItemText))
          primary="Blog"
          secondary="https://www.example-blog.com"
        />
      </WithStyles(ForwardRef(ListItem))>
    </WithStyles(ForwardRef(List))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
